Editor/Analytics/AnalyticsUtils.cs

```diff
@@ -20,32 +20,54 @@ internal static AnalyticsState GetTrackingState()
             {
                 return AnalyticsState.Disabled;
             }
-            return EditorPrefExtensions.GetEnum(KeyTrackingState, AnalyticsState.Unknown);
+
+            try
+            {
+                return EditorPrefExtensions.GetEnum(KeyTrackingState, AnalyticsState.Unknown);
+            }
+            catch (ArgumentException)
+            {
+                EditorPrefExtensions.SetEnum(KeyTrackingState, AnalyticsState.Unknown);
+                return AnalyticsState.Unknown;
+            }
         }
 
         internal static void SetTrackingTo(AnalyticsState state)
         {
-            if (state == AnalyticsState.Minimal)
+            AnalyticsState currentState = GetTrackingState();
+            switch (state)
             {
-                // Without a stored session id a random one will be created every time.
-                if (EditorPrefs.HasKey(BaseAnalyticsTracker.KeySessionId))
-                {
-                    EditorPrefs.DeleteKey(BaseAnalyticsTracker.KeySessionId);
-                }
-            }
+                case AnalyticsState.Enabled when currentState == AnalyticsState.Disabled:
+                    string id = EditorPrefs.GetString(BaseAnalyticsTracker.KeySessionId, null);
+                    if (string.IsNullOrEmpty(id))
+                    {
+                        // Create new session id, which allows better tracking
+                        EditorPrefs.SetString(BaseAnalyticsTracker.KeySessionId, Guid.NewGuid().ToString());
+                    }
+                    EditorPrefExtensions.SetEnum(KeyTrackingState, AnalyticsState.Enabled);
+                    SendTrackingEvent(trackingEnabled: true);
+                    break;
 
-            if (state == AnalyticsState.Enabled)
-            {
-                string id = EditorPrefs.GetString(BaseAnalyticsTracker.KeySessionId, null);
-                if (string.IsNullOrEmpty(id) || id.StartsWith("IA") == false)
-                {
-                    // Create new session id starting with IA, which allows better tracking
-                    EditorPrefs.SetString(BaseAnalyticsTracker.KeySessionId,
-                        "IA" + Guid.NewGuid().ToString().Substring(2));
-                }
+                case AnalyticsState.Disabled when currentState == AnalyticsState.Enabled:
+                    SendTrackingEvent(trackingEnabled: false);
+                    EditorPrefExtensions.SetEnum(KeyTrackingState, AnalyticsState.Disabled);
+                    break;
+
+                default:
+                    EditorPrefExtensions.SetEnum(KeyTrackingState, state);
+                    break;
             }
+        }
 
-            EditorPrefExtensions.SetEnum(KeyTrackingState, state);
+        private static void SendTrackingEvent(bool trackingEnabled)
+        {
+            string label = trackingEnabled ? "enabled" : "disabled";
```

Comment:
Cant this be in the line with the Label = assignment? Also probably just use the enum instead of a bool and ToString it.